### ISSUE TITLE
Purge deletes player passwords with bypass permission

### ISSUE THREAD

**Issue by DarkFort** [AuthMe/AuthMeReloaded#1716]
AuthMe deletes the passwords of players who have authme.bypasspurge permission.
I tried all versions of AuthMe, the problem was not solved.
Latest LuckPerms

**Comment by MrMIStakee** [comment]
The same problem!

/authme purge - removes players with the right “authme.bypasspurge”

I am using the latest version of LuckPerms

**Comment by andrey433** [comment]
The same problem!

**Comment by elsol42** [comment]
I tried to solve this problem for 4 days, I thought it was just my problem.
![o0omg2ixnxi](https://user-images.githubusercontent.com/46028846/50294761-aa8d6500-0487-11e9-931d-7ff4a4f4286b.jpg)

**Comment by Diarg0ne** [comment]
Yes, it doesn't work with Luckperms.

**Comment by MrLaco** [comment]
Unfortunatelly, I have the same problem too. What can I do to solve it?

**Comment by sgdc3** [comment]
yeah purging is still broken with luckperms, we have to figure out how to fix it properly

**Comment by ProblemsSender** [comment]
Nah, actually I'm using Pex, it means this bug is pretty old... so it must from AuthMe conflict with itself function, or with Essentials? We should test it to find out what is the problem.

**Comment by sgdc3** [comment]
@ProblemsSender i think your is a different issue

**Comment by ljacqu** [comment]
Relates to #1544

**Comment by sgdc3** [comment]
Pex has a different issue, the LuckPerms issue is that it is fully async and it has to load data of offline players! AuthMe purge system expects sync permission lookups, but when we ask LP if the user has the permission the data isn't loaded yet!

**Comment by sgdc3** [comment]
@ljacqu the only way to fix that is to rewrite the purge system to support async permission lookups
